cEP-0009.md

```diff
@@ -0,0 +1,181 @@
+| Metadata     |                                                          |
+| ------------ |----------------------------------------------------------|
+| cEP          | 9                                                        |
+| Version      | 0.1                                                      |
+| Title        | Extract and utilize information from project files       |
+| Authors      | Satwik Kansal <satwikkansal@gmail.com>                   |
+| Status       | Proposed                                                 |
+| Type         | Feature                                                  |
+
+## Abstract
+
+There are many kind of project files like ".editorconfig", "Gruntfile.js",
+etc. that contain meta-information that can used by tools like
+coala-quickstart to provide better analysis. This cEP proposes a mechnaism to
+represent and extract information from project files and an interface to utilize the extracted information.
+
+## Proposed Approach
+
+### 1. Implementation of the “Info” class
+
+A subclass of `Info` class will represent a specific kind of information. For example, the class `IgnorePathInfo` may represent “ignored file paths”.
+
+```py
+class Info:
+    description = "Some information"
+    # for validation
+    value_type = "data-type of the information value, can also be an ``Enum``"
+                 "of possible choices."
+
+    def __init__(self,
+                 source,
+                 value: value_type):
+        self.source = source
+        self.value = value
+        self.name = self.__class__.__name__
```

Comment:
Do you want this to be inherited? Like if `class IgnoredPathsInfo(Info):` is defined, do you want the `.name` attribute to be available there? If so, you'll need to define it as a `__getattr__` or use a `@property`.

If not, ignore me :P